plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java

```diff
@@ -627,6 +643,20 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         return Optional.of("(" + String.join(",", prop) + ")");
     }
 
+    private static ObjectWriteFunction uInt64WriteFunction()
+    {
+        return ObjectWriteFunction.of(
+                Int128.class,
+                (statement, index, value) -> {
+                    BigInteger unscaledValue = value.toBigInteger();
+                    BigDecimal bigDecimal = new BigDecimal(unscaledValue, UINT64_TYPE.getScale(), new MathContext(UINT64_TYPE.getPrecision()));
+                    if (bigDecimal.compareTo(UINT64_MIN_VALUE) < 0 || bigDecimal.compareTo(UINT64_MAX_VALUE) > 0) {
```

Comment:
`// ClickHouse stores incorrect results when the values are out of supported range.` ?

Reply:
Maybe same for others too.